google-cloud-storage/lib/google/cloud/storage/bucket.rb

```diff
@@ -1169,10 +1169,22 @@ def update if_metageneration_match: nil, if_metageneration_not_match: nil
           updater.check_for_changed_labels!
           updater.check_for_mutable_cors!
           updater.check_for_mutable_lifecycle!
-          return if updater.updates.empty?
-          patch_gapi! updater.updates,
-                      if_metageneration_match: if_metageneration_match,
-                      if_metageneration_not_match: if_metageneration_not_match
+
+          attributes = Array(updater.updates)
```

Comment:
Consider creating a helper method `update_gapi!` to parallel the existing `patch_gapi!`. That will make it easier to see how this code relates to the existing code, clarifying things for potential future refactors.

Reply:
Thanks @dazuma. I added the helper method. Now the flow of the code is:
```
bucket.update --> bucket.update_gapi! --> service.update_bucket --> apiary.update_bucket
```